### DISCUSSION TITLE
Feature suggestion: allow structured values (of an irrefutable shape) in specification header

### DISCUSSION THREAD

**Discussion by fpottier** [ocaml-gospel/gospel#151] [Ideas]
In the following `.mli` file, the first declaration is valid Gospel but ugly. The second declaration seems much more natural but seems currently invalid: it is rejected by `gospel check` and by Cameleer. Would it be possible to allow it? This would seemingly make sense at least for tuples and records.
```
val write : int array * int array -> unit
(*@ write ab
    requires Array.length (snd ab) > 0 *)

val write2 : int array * int array -> unit
(*@ write2 (a, b)
    requires Array.length b > 0 *)
```

**Comment by pascutto** [comment]
I agree with this. I've created #153 for this. This is somewhat related to #129 too.
